edx/edx-enterprise
# [ENT-496] Migrate to new DataSharingConsent model

**Description:** This PR migrates the codebase to use the new `DataSharingConsent` model created in #167. It also introduces a data migration in order to make that code migration meaningful, and this data migration comes with tests to build confidence.

**JIRA:** [ENT-496](https://openedx.atlassian.net/browse/ENT-496)

**Dependencies:** Not a hard dependency because I've already incorporated the changes and expect them to be merged soon -- #158.

**Merge deadline:** Before next sprint starts, hopefully

**Testing instructions:**

1. Ensure that all enterprise flows work fine as before.
1. Ensure that data sharing consent screens properly render, or don't.
1. Ensure that data sharing consent is properly saved.
1. Ensure that any management commands that depend on consent (like `transmit_learner_data`) continue to filter properly.
1. Ensure that the new `consent` application continues to work as intended (i.e. send some requests to the Consent API endpoint).

Since this touches so many parts of the codebase, it's hard to be specific here.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** Since we're keeping around the UDSCA and ECE models still, I've at least put up deprecation warnings on using their functionality. I've ignored those warnings in certain places where they became spammy, and if we do decide to get rid of these models some day, those suppressions will obviously have to be removed.
